src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1286,16 +1286,21 @@
 (defn update-latest-report!
   "Given a node name, updates the `certnames` table to ensure that it indicates the
    most recent report for the node."
-  [node]
-  {:pre [(string? node)]}
-  (let [latest-report (:id (first (query-to-vec
-                                    ["SELECT id FROM reports
+  [node report-id report-timestamp]
+  {:pre [(string? node) (integer? report-id)]}
```

Comment:
I wanted to validate the report-timestamp coming in but I'm not sure what datatype it is to check.